scripts/send_spans.py

```diff
@@ -0,0 +1,427 @@
+    datetime_fields = {}
+    if add_datetimes:
+        datetime_fields = {
+            "start_datetime": start_timestamp.isoformat(),
+            "end_datetime": end_timestamp.isoformat(),
+        }
+    return {
+        "event_id": uuid.uuid4().hex,
+        "organization_id": ORGANIZATION_ID,
+        "project_id": project_id,
+        "trace_id": trace_id,
+        "span_id": uuid.uuid4().hex[:16],
+        "parent_span_id": parent_span_id,
+        "segment_id": segment_id,
+        "profile_id": uuid.uuid4().hex,
+        "is_segment": is_segment,
+        "start_timestamp_ms": int(start_timestamp.timestamp() * 1000),
+        "start_timestamp_precise": start_timestamp.timestamp(),
+        "end_timestamp_precise": end_timestamp.timestamp(),
+        "duration_ms": duration_ms,
+        "exclusive_time_ms": duration_ms,
+        "retention_days": 90,
+        "received": RECEIVED,
+        "description": "This is a span",
+        "tags": custom_tags(project_id),
+        "sentry_tags": sentry_tags(transaction_name),
+        "measurements": {
+            "measurement1": {
+                "value": random.random(),
+                "unit": "unit",
+            },
+            "measurement2": {
+                "value": random.randint(1, 100000),
+                "unit": "unit",
+            },
+        },
+        **datetime_fields,
+    }
+
+
+def update_span_duration(span: Span, duration_ms: int) -> Span:
+    start_timestamp = datetime.datetime.fromtimestamp(span["start_timestamp_precise"])
+    end_timestamp = start_timestamp + datetime.timedelta(milliseconds=duration_ms)
+    span["end_timestamp_precise"] = end_timestamp.timestamp()
+    span["duration_ms"] = duration_ms + 10
+    span["exclusive_time_ms"] = 10
+    return span
+
+
+def generate_span_branch(
+    trace_id: str,
+    project_id: int,
+    start_timestamp: datetime.datetime,
+    transaction_name: str,
+    parent_span_id: str,
+    span_count: int = 0,
+    prefix: str = "",
+    verbose: bool = False,
+    add_datetimes: bool = False,
+) -> tuple[list[Span], int]:
+    duration = 0
+    spans = []
+    for i in range(randint(3, 10)):
+        span_duration = randint(1, 50)
+        span = create_span(
+            trace_id=trace_id,
+            parent_span_id=parent_span_id,
+            segment_id=transaction_name,
+            is_segment=False,
+            project_id=project_id,
+            duration_ms=span_duration,
+            start_timestamp=start_timestamp + datetime.timedelta(milliseconds=duration),
+            transaction_name=transaction_name,
+            add_datetimes=add_datetimes,
+        )
+
+        to_split_or_not_to_split = random.random()
+        if span_count < 100 and to_split_or_not_to_split < 0.3:
+            subspans = []
+            subduration = 0
+            if project_id != 3 and to_split_or_not_to_split < 0.15:
+                if verbose:
+                    print(f"{prefix}T{project_id}")
```

Comment:
I feel like log level is a better way to do this

Reply:
I agree, I just stopped improving this basically.